.eslintrc

```diff
@@ -101,5 +101,43 @@
     },
     "parserOptions": {
         "ecmaVersion": 2018
-    }
+    },
+    "overrides": [
+        {
+            "files": ["test/**"],
+            "env": {
+                "mocha": true,
+                "es2020": true
+            },
+            "plugins": [
+                "mocha"
+            ],
+            "extends": [
+                "plugin:mocha/recommended"
+            ],
+            "rules": {
+                "mocha/no-hooks-for-single-case": "off",
+                "mocha/no-setup-in-describe": "off",
+                "camelcase": "off",
+                "max-len": ["warn", 130],
+                "mocha/prefer-arrow-callback": ["error", {
+                    "allowNamedFunctions": true
+                }],
+                "padded-blocks": ["error", "never"]
+            },
+            "parserOptions": {
+                "ecmaVersion": 11
+            }
+        },
+        {
+            "files": ["code_samples/**/*"],
+            "env": {
+                "es2020": true
+            },
+            "parserOptions": {
+                "ecmaVersion": 11
+            }
+        }
+    ],
+    "ignorePatterns": ["ts.eslintrc.js"]
```

Comment:
Does this have some special meaning? I don't see a file called `ts.eslintrc.js`

Reply:
Ah sorry I created it and deleted it, fixing..

Reply:
Yes, I mean they are not hugely different, if there is no special reason for the difference, I would just use what is defined on the top level